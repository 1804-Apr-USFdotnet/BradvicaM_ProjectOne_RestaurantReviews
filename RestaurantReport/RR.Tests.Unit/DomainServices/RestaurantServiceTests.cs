using System;
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using RR.DomainServices;
using RR.Models;
using RR.RepositoryContracts;

namespace RR.Tests.Unit.DomainServices
{
    [TestClass]
    public class RestaurantServiceTests
    {
        private readonly RestaurantService _restaurantService;
        private readonly Mock<IRestaurantRepository> _mockRepository;

        public RestaurantServiceTests()
        {
            _mockRepository = new Mock<IRestaurantRepository>();
            _mockRepository.Setup(x => x.GetById(It.IsAny<Guid>())).Returns(new Restaurant{Reviews = new List<Review>()});
            _mockRepository.Setup(x => x.Update(It.IsAny<Restaurant>()));

            _restaurantService = new RestaurantService(_mockRepository.Object);
        }

        [TestMethod]
        public void ReviewRestaurant_GivenReview_CallsRepositoryUpdate()
        {
            var review = new Review{Restaurant = new Restaurant()};

            _restaurantService.ReviewRestaurant(review);

            _mockRepository.Verify(x => x.Update(It.IsAny<Restaurant>()), Times.AtLeastOnce);
        }

        [TestMethod]
        public void Get_GivenRestaurantId_CallRepositoryMethod()
        {
            _restaurantService.Get(Guid.NewGuid());

            _mockRepository.Verify(x => x.GetById(It.IsAny<Guid>()), Times.AtLeastOnce);
        }
    }
}
