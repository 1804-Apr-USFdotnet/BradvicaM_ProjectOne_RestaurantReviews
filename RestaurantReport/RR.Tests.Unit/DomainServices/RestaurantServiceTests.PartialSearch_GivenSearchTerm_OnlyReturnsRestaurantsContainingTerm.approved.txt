Restaurants:[0] = 
Id: 00000000-0000-0000-0000-000000000000
RestaurantId: 00000000-0000-0000-0000-000000000000
Name: Mike
Street: 123 billy St.
City: Tampa
State: FL
ZipCode: 18172
PhoneNumber: 123412312
Website: www.hithere.com
Rating: 9

Restaurants:[1] = 
Id: 00000000-0000-0000-0000-000000000000
RestaurantId: 00000000-0000-0000-0000-000000000000
Name: Frank
Street: 123 Me St.
City: Tampa
State: billy
ZipCode: 18172
PhoneNumber: 123412312
Website: www.hithere.com
Rating: 4

